internal/subscriber/consumer.go

```diff
@@ -47,12 +48,17 @@ type consumerManager struct {
 // NewConsumerManager ...
 func NewConsumerManager(ctx context.Context, bs brokerstore.IBrokerStore, brokerTimeout int, subscriberID string, subscriptionName,
 	topicName, retryTopicName string) (IConsumer, error) {
+	autoOffsetReset := "latest"
+	if tp.IsDLQTopic(topicName) {
+		autoOffsetReset = "earliest"
```

Comment:
Every time a new consumer is instantiated for a DLQ topic, it will retry messages from the start of the topic. Ideally, the offset must be set once during the time of DLQ subscription creation.

Reply:
auto offset reset is only used if there is no committed offset. Even if consumer dies, and next consumer will start from last committed offset. In case there is no committed offset, means it is the first time we are consuming from the topic for that consumer group, then only this config will be used.
https://stackoverflow.com/questions/32390265/what-determines-kafka-consumer-offset